.changeset/stupid-glasses-switch.md

```diff
@@ -0,0 +1,40 @@
+---
+'astro': minor
+---
+
+Adds a new virtual module called `astro:config` as part of the relative [RFC Stage 3](https://github.com/withastro/roadmap/pull/1106).
+
+The new virtual module exposes a type-safe sub-set configuration defined in `astro.config.mjs`.
+
+The virtual module exposes two sub-paths:
+- `astro:config/client`: exposes config information that are safe to expose to the client.
+- `astro:config/server`: exposes additional information that is safe to expose to the server, such as file/dir paths.
+
+To enable this new virtual module, an experimental flag must be turned on:
+
+```js
+// astro.config.mjs
+import {defineConfig} from "astro/config"
+export default defineConfig({
+  experimental: {
+    serializeManifest: true
+  }
+})
+```
+
+Then, you can use the module in any file inside the Astro project
+
+```js
+// src/utils.js
+import { trailingSlash } from "astro:config/client";
+
+function addForwardSlash(path) {
+  if (trailingSlash === "always") {
+    return path.endsWith("/") ? path : path + "/"
+  } else {
+    return path
+  }
+}
+
+```
+
```

Comment:
```suggestion

For a complete overview, and to give feedback on this experimental API, see the [Serialized Manifest RFC](https://github.com/withastro/roadmap/blob/feat/serialised-config/proposals/0051-serialized-manifest.md).
```

Noted in adding this standard line, the Stage 3 PR 1106 hasn't yet been added to the description there.